backend/de.metas.shipper.client.nshift/src/main/java/de/metas/shipper/client/nshift/NShiftShipAdvisorService.java

```diff
@@ -127,22 +130,34 @@ private static JsonDeliveryAdvisorResponse buildJsonDeliveryResponse(@NonNull fi
 		final JsonShipAdvisorResponseProduct product = response.getProducts().get(0);
 		final JsonDeliveryAdvisorResponse.JsonDeliveryAdvisorResponseBuilder responseBuilder = JsonDeliveryAdvisorResponse.builder()
 				.requestId(requestId)
-				.shipperProduct(product.getProdName())
-				.responseItem(NShiftConstants.PROD_CONCEPT_ID, String.valueOf(product.getProdConceptID()));
+				.shipperProduct(JsonShipperProduct.builder()
+						.name(product.getProdName())
+						.code(String.valueOf(product.getProdConceptID()))
+						.build());
 
 		final JsonShipAdvisorResponseGoodsType productGoodsType = product.getProductGoodsType();
 		if (productGoodsType != null)
 		{
-			responseBuilder.responseItem(NShiftConstants.GOODS_TYPE_ID, String.valueOf(productGoodsType.getGoodsTypeId()));
-			responseBuilder.responseItem(NShiftConstants.GOODS_TYPE_NAME, productGoodsType.getGoodsTypeName());
+			responseBuilder.goodsType(JsonGoodsType.builder()
+					.id(String.valueOf(productGoodsType.getGoodsTypeId()))
+					.name(productGoodsType.getGoodsTypeName())
+					.build());
 		}
 
-		responseBuilder.shipperProductServices(product.getServices().stream()
-				.map(JsonShipAdvisorResponseService::getServiceId)
-				.map(String::valueOf)
+		responseBuilder.shipperProductServices(product.getServices()
+				.stream()
+				.map(NShiftShipAdvisorService::toJsonCarrierService)
 				.collect(ImmutableList.toImmutableList())
 		);
 
 		return responseBuilder.build();
 	}
+
+	private static JsonCarrierService toJsonCarrierService(@NonNull final JsonShipAdvisorResponseService jsonShipAdvisorResponseService)
```

Comment:
@adi-stefan  @mrie04  i can spot there will be hard to distinguish between generic/common API JSONs and NShift JSONs.
we have to think how to make it more clear/obvious.